File: signal_new/CCA_analysis/standard_CCA_ITCCA.py
import numpy as np
from standard_CCA import standard_cca
from IT_CCA import it_cca


def standard_cca_it_cca(signal, sampling_rate, fund_frequency, num_harmonics, template):
    corr1, Wxy, _, _ = standard_cca(signal, sampling_rate=sampling_rate, fund_frequency=fund_frequency,
                                    num_harmonics=num_harmonics)
    corr5, Wx, _ = it_cca(signal, template)
    corr2 = np.corrcoef(signal.dot(Wx).T, template.dot(Wx).T)
    corr3 = np.corrcoef(signal.dot(Wxy).T, template.dot(Wxy).T)
    _, Wxxy, _, _ = standard_cca(template, sampling_rate=sampling_rate, fund_frequency=fund_frequency,
                                 num_harmonics=num_harmonics)
    corr4 = np.corrcoef(signal.dot(Wxxy).T, template.dot(Wxxy).T)
    rho_array = np.array([corr1, corr2[0, 1], corr3[0, 1], corr4[0, 1], corr5])
    r = np.sum(np.sign(rho_array) * np.power(rho_array, 2))
    return r

File: signal_new/CCA_analysis_matlab/standardCCA_ITCCA.m
function r = standardCCA_ITCCA(Signal, fundFrequency, numHarmonics, Template)
[r1, Wxy, ~, ~] = standardCCA(Signal, fundFrequency, numHarmonics);
[r5, Wx, ~] = IT_CCA(Signal, Template);
r2 = corrcoef([Signal*(Wx(:,1)) Template*(Wx(:,1))]);
r3 = corrcoef([Signal*(Wxy(:,1)) Template*(Wxy(:,1))]);
[~, Wxxy, ~, ~] = standardCCA(Template, fundFrequency, numHarmonics);
r4 = corrcoef([Signal*(Wxxy(:,1)) Template*(Wxxy(:,1))]);
rho_array = zeros(1,5);
rho_array(1)=r1(1);
rho_array(2)=r2(1,2);
rho_array(3)=r3(1,2);
rho_array(4)=r4(1,2);
rho_array(5)=r5(1);
r=sum(sign(rho_array).*power(rho_array,2));
end